_posts/2019-04-03-blog-import-the-style-to-the-typescript-file.md

```diff
@@ -0,0 +1,112 @@
+categories: blog
+author: Jeff Wu
+---
+
+![](/integration-blog/assets/2019-04-03-blog-import-the-style-to-the-typescript-file/babel-ts-scss.jpg)
+
+### TypeScript show the error when import style file like scss or css
+
+After the success of migrating to TypeScript development from JavaScript,  we can get more confident in building an incredible product. But the challenge will not stop. When I import a style file to TypeScript file, the editor shows an error said that could not find the module. To overcome the issue about the style file importing is the next challenge we need to face.
+
+I am researching the solution to avoid the problem that the error message showed due to the TS compile do not know about that when importing the style file. Currently, I found some solution and need to clarify which one is the better choice.
+
+
+
+### Using the "typings-for-css-modules-loader"
+It's a Webpack loader make our style file like SCSS can have intellisense by generating a TypeScript definition (d.ts). If you have using the "css-loader" you can easily set up the Webpack configuration by the following:
+
+```javascript
+{
+  loader: 'typings-for-css-modules-loader',
+    options: {
+      modules: true,
+      localIdentName: `${hashPrefix}_[path]_[name]_[local]_[hash:base64:5]`,
+      namedExport: true,
+      camelCase: true,
+			banner: '// *** Generated File - Do not Edit ***'
+    }
+}
+```
+However, it has some flaw. The process of the definition file generator need always start a watch by a command like `yarn start`. It also makes our size of the repository become a bigger one although the intellisense is convenient for us.
+
+
+
+### Using the "require" syntax instead of using import.
+
+Using the different style or syntax for import may make our project become a mess, we should keep a consistent form of the import, so maybe there were have a better solution that I not found yet.
+
+
+
+### TypeScript server plugin - [typescript -plugin-css-modules](https://github.com/mrmckeb/typescript-plugin-css-modules)
+This plugin is a TypeScript plugin was inspired by this [create-react-app issue](https://github.com/facebook/create-react-app/issues/5677) and based on css-module-types. It's easy to set up by installing the node module and add plugin to the `tsconfig.json` like the following:
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [{ "name": "typescript-plugin-css-modules" }]
+  }
+}
+```
+
+The match pattern default is `\\.module\\.(sa|sc|c)ss$`. We can custom set other pattern by the property.
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "typescript-plugin-css-modules",
+        "options": {
+          "customMatcher": "\\.m\\.css$",
+          "camelCase": "dashes"
+        }
+      }
+    ]
+  }
+}
+```
+After the setting, we also have to change the version of TypeScript. By default, VSCode will use its own version of TypeScript. To make it work correctly with this plugin, there have two options:
```

Comment:
After configurations, we also have to change the version of TypeScript used by VSCode. By default, VSCode comes with its own version of TypeScript. However, we can specify the TS version by two different ways: